fbw-a32nx/src/wasm/extra-backend/CMakeLists.txt

```diff
@@ -0,0 +1,155 @@
+
+# add all cpp source-files here
+set(SOURCES
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/MsfsHandler.cpp
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataManager.cpp
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes/CacheableVariable.cpp
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes/NamedVariable.cpp
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes/AircraftVariable.cpp
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes/ClientEvent.cpp
+        src/Gauge_Extra_Backend.cpp
+        src/Example/ExampleModule.cpp
+        src/LightingPresets/LightingPresets.cpp
+        src/Pushback/Pushback.cpp
+        src/Pushback/InertialDampener.cpp
+        src/AircraftPresets/AircraftPresets.cpp
+        )
+# add all include directories here
+set(INCLUDES
+        "${MSFS_SDK}/WASM/include"
+        "${MSFS_SDK}/WASM/wasi-sysroot/include"
+        "${MSFS_SDK}/WASM/wasi-sysroot/include/c++/v1"
+        "${MSFS_SDK}/SimConnect SDK/include"
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/lib
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler
+        ${FBW_ROOT}/fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes
+        src
+        src/Modules
+        src/Example
+        src/LightingPresets
+        src/Pushback
+        src/AircraftPresets
+        )
+
+# This builds the object library - that is all .o files
+# the resulting .o files are stored in $<TARGET_OBJECTS:${PROJECT_NAME}-lib>
+add_library(${PROJECT_NAME}-lib OBJECT ${SOURCES})
+target_include_directories(${PROJECT_NAME}-lib PUBLIC ${INCLUDES})
+
+# wasm-ld command
+set(WASM_LD_CMD "wasm-ld")
+set(WASM_LD_OUTPUT_FLAG "-o ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.wasm")
+set(WASM_LD_FLAGS "${WASM_LD_ARGS}")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} --no-entry --allow-undefined")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} --export __wasm_call_ctors --export-dynamic --export malloc --export free --export-table")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} --gc-sections")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} -lc++ -lc++abi")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} -L \"${MSFS_SDK}/WASM/wasi-sysroot/lib/wasm32-wasi\"")
+set(WASM_LD_FLAGS "${WASM_LD_FLAGS} -lc \"${MSFS_SDK}/WASM/wasi-sysroot/lib/wasm32-wasi/libclang_rt.builtins-wasm32.a\"")
+set(WASM_LD_FULL_CMD "${WASM_LD_CMD} ${WASM_LD_OUTPUT_FLAG} ${WASM_LD_FLAGS}")
+
+# wasm-opt command to optimize the wasm file
+set(WASM_OPT_CMD "wasm-opt")
+set(WASM_OPT_FLAGS "-O1 -o") # anything higher than -O1 will break the wasm file for the sim
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(WASM_OPT_FULL_CMD "cp ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.wasm ${OUT_PATH} ")
+else ()
+    set(WASM_OPT_FULL_CMD "${WASM_OPT_CMD} ${WASM_OPT_FLAGS} ${OUT_PATH} ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.wasm")
+endif ()
+
+# This builds the wasm file
+add_custom_command(OUTPUT always-build-wasm
```

Comment:
After the cpp compile the wasm needs to be build (linked) every time - otherwise it doesn't even make sense ;)

If someone finds a better and reliable solution I'll take it. E.g. I tried to actually use WASM LD like a normal linker but wasn't successful. 

The code is now incrementally build in parallel - that was the whole point of using CMake and Ninja. 
Before with the build.sh scripts it was always everything and serial - much faster now esp. if the code is actually spread in compile units. If everything is in .h and .hpp files then of course there isn't much benefit. 

And the wasm-ld and wasm-opt steps are rather fast so wasn't an issue until now - haven't tested it with larger code base but at least for fbw.wasm these steps were already fast enough with the sh script.